### PR TITLE
Plugins: Track SystemJS load errors with Faro

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Adds an error handler for the `SystemJS.import` call in `importPluginModule`, which reports the loading error to Faro.

This is useful to keep track of frontend plugin loading errors, and can be used to build dashboards like this one: https://ops.grafana-ops.net/d/fe0d26cpnaqkga/plugins-cdn-sri-checks-frontend-observability

**Why do we need this feature?**

Track plugin load errors in ops via Faro, especially for instances with `pluginsSriChecks` feature toggle enabled.

**Who is this feature for?**

Grafana Labs.

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana-community-team/issues/15

**Special notes for your reviewer:**

The ephemeral instance https://ephemeral1511182194465xnyo.grafana-dev.net can be used to test the feature.

This instance has plugins CDN (`cdnPlugins` feature toggle) and SRI checks enabled (`pluginsSriChecks` feature toggle)

That instance is running commit a17d2865ae17fb3b8b5b305e52021ac1e627d800, which also breaks all module.js hashes from the backend (reverted in commit 02a61f4046772032c3391b497668147dd0039b0a).

**In other words,  SRI checks will ALWAYS fail for that instance.**

This dashboard can be used to track plugin load errors: https://ops.grafana-ops.net/d/fe0d26cpnaqkga/plugins-cdn-sri-checks-frontend-observability

If you target the "dev" cluster, you should also see data coming from the Ephemeral instance.

I did some testing between 2024-10-09T14:53:59.304Z and 2024-10-09T15:31:06.912Z on the ephemeral instance, so you should be able to see some data in there.

Direct link for the dashboard with that time interval applied: https://ops.grafana-ops.net/d/fe0d26cpnaqkga/plugins-cdn-sri-checks-frontend-observability?from=2024-10-09T14:53:59.304Z&to=2024-10-09T15:31:06.912Z&timezone=utc&var-query0=c11b2b62-5b3d-4012-bc2a-0d59e9ec8942&var-loki_ds=grafanacloud-logs&var-query0-2=&var-environment=Hosted%20Grafana%20-%20Dev&var-query0-3=&var-prometheus_ds=000000134&showCategory=Panel%20options

Explore link with sample query and sample results: https://ops.grafana-ops.net/goto/kzYvsXzNR?orgId=1

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by academo — APPROVED** [review]

File: public/app/features/plugins/plugin_loader.ts [hunk 1]
```diff
@@ -118,7 +119,18 @@ export async function importPluginModule({
...
+    console.error(error);
+    pluginsLogger.logError(error, {
+      path,
+      pluginId,
+      pluginVersion: version ?? '',
+      expectedHash: moduleHash ?? '',
+      sriChecksEnabled: (config.featureToggles.pluginsSriChecks ?? false).toString(),
+    });
+    throw error;
+  });
```
Comment (jackw) [inline public/app/features/plugins/plugin_loader.ts:133]: Are we happy to only track usage of  `importPluginModule` or do want to track all usage of `System.import`?

If it's the latter we likely want to extend the System prototype import and move this logging there. If we only care about module.js then we can ignore this comment entirely. 😄

File: public/app/features/plugins/plugin_loader.ts [hunk 2]
```diff
@@ -118,7 +119,18 @@ export async function importPluginModule({
...
-  return SystemJS.import(modulePath);
+  return SystemJS.import(modulePath).catch((e) => {
+    let error = new Error('Could not load plugin: ' + e);
+    console.error(error);
+    pluginsLogger.logError(error, {
+      path,
+      pluginId,
+      pluginVersion: version ?? '',
+      expectedHash: moduleHash ?? '',
+      sriChecksEnabled: (config.featureToggles.pluginsSriChecks ?? false).toString(),
```
Comment (jackw) [inline public/app/features/plugins/plugin_loader.ts:130]: Do we want to track `loadingStrategy` here too? 🤔

**Review by jackw — APPROVED** [review]
Great stuff @xnyo . 🚀 

Left a couple of comments, feel free to ignore them if you like.

File: public/app/features/plugins/plugin_loader.ts (hunk 2 above)
Comment (xnyo) [reply public/app/features/plugins/plugin_loader.ts:131]: Good point, I've added it! :+1:

**Pull request merged**